ambari-server/src/main/java/org/apache/ambari/server/state/ConfigHelper.java

```diff
@@ -1743,15 +1743,22 @@ public boolean checkStaleConfigsStatusForHostComponent(ServiceComponentHost sch,
       String type = changedConfigType.getKey();
       stale |= (serviceInfo.hasConfigDependency(type) || componentInfo.hasConfigType(type));
       if (stale) {
-        changedProperties.addAll(changedConfigType.getValue());
+        for (String propertyName : changedConfigType.getValue()) {
+          changedProperties.add(type + "/" + propertyName);
+        }
       }
     }
 
     String refreshCommand = calculateRefreshCommand(stackInfo.getRefreshCommandConfiguration(), sch, changedProperties);
 
+    Map<String, HostConfig> actual = sch.getActualConfigs();
+
+    Map<String, Map<String, String>> desired = getEffectiveDesiredTags(cluster, sch.getHostName(),cluster.getDesiredConfigs());
+
```

Comment:
nit: any logs could help improve debugging?

Reply:
> nit: any logs could help improve debugging?

@virajjasani 
Logging alone may not be enough, but I recommend using remote debug mode.

Reply:
i understand, but i was just curious if you think we might be able to improve some logging here for debugging in prod env, but the changes look good, please feel free to merge the PR. Thanks

Reply:
Yeah, that's a great idea. But I found logs missing all the way down the data link, not just this one. Maybe we should start a discussion to improve Ambari's logging？
Thx.